File: backend/src/AsliApp.Domain/Education/Quiz.cs
namespace AsliApp.Domain.Education;

public sealed class Quiz
{
    public Guid Id { get; set; }
    public Guid LessonId { get; set; }
    public string Title { get; set; } = string.Empty;
    public bool IsPublished { get; set; }
    public DateTimeOffset CreatedAtUtc { get; set; }
    public DateTimeOffset UpdatedAtUtc { get; set; }

    public Lesson Lesson { get; set; } = null!;
    public ICollection<QuizQuestion> Questions { get; set; } = [];
    public ICollection<QuizAttempt> Attempts { get; set; } = [];
}

File: mobile/lib/features/quiz/domain/models/quiz.dart
import 'package:asli_app/features/quiz/domain/models/quiz_question.dart';

final class Quiz {
  Quiz({
    required this.id,
    required this.lessonId,
    required this.title,
    required List<QuizQuestion> questions,
  }) : assert(questions.isNotEmpty),
       questions = List.unmodifiable(questions);

  final String id;
  final String lessonId;
  final String title;
  final List<QuizQuestion> questions;
}
